src/pip/_internal/models/direct_url.py

```diff
@@ -64,18 +65,13 @@ def _filter_none(**kwargs: Any) -> Dict[str, Any]:
     return {k: v for k, v in kwargs.items() if v is not None}
 
 
+@dataclass
 class VcsInfo:
```

Comment:
I opted to leave the direct_url classes non-frozen as some tests broke and it would be more consistent with `ArchiveInfo` which I did not convert (due to non-trivial logic I didn't feel like groking).